Mage.Sets/src/mage/cards/g/GethThaneOfContracts.java

```diff
@@ -0,0 +1,98 @@
+package mage.cards.g;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.ActivateAsSorceryActivatedAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.CompositeCost;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.abilities.effects.common.ReturnFromGraveyardToBattlefieldTargetEffect;
+import mage.abilities.effects.common.continuous.BoostControlledEffect;
+import mage.abilities.effects.common.continuous.GainAbilityTargetEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.events.ZoneChangeEvent;
+import mage.target.common.TargetCardInYourGraveyard;
+
+import java.util.UUID;
+
+public class GethThaneOfContracts extends CardImpl {
+    public GethThaneOfContracts(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{B}{B}");
+        this.addSuperType(SuperType.LEGENDARY);
+        this.addSubType(SubType.PHYREXIAN);
+        this.addSubType(SubType.ZOMBIE);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(4);
+
+        //Other creatures you control get -1/-1.
+        this.addAbility(new SimpleStaticAbility(
+                new BoostControlledEffect(-1, -1, Duration.WhileOnBattlefield, true)
+        ));
+
+        //{1}{B}{B}, {T}: Return target creature card from your graveyard to the battlefield. It gains “If this creature
+        //would leave the battlefield, exile it instead of putting it anywhere else.” Activate only as a sorcery.
+        Ability ability = new ActivateAsSorceryActivatedAbility(
+                new ReturnFromGraveyardToBattlefieldTargetEffect(false, false)
+                        .setText("Return target creature card from your graveyard to the battlefield."),
+                new CompositeCost(new ManaCostsImpl<>("{1}{B}{B}"), new TapSourceCost(), "{1}{B}{B}, {T}")
```

Comment:
This doesn't need to be a composite cost, as it does not cooperate with xmage's purple-border UI indicator for showing cards with abilities that can be activated (in other words, this card shows as purple-bordered even when the controller does not have the means to produce {1}{B}{B}). Additionally, this being a composite cost results in an ability choice menu when selecting the card despite it only having one activated ability.

Instead, you should make the cost for `ActivateAsSorceryActivatedAbility` a simple `new ManaCostsImpl<>("{1}{B}{B}")`,  then write `ability.addCost(new TapSourceCost());` on a new line

Reply:
good to know... didn't know that there is an addCost method xD